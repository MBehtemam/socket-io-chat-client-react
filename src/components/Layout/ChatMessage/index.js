import styled from "styled-components";

const ChatMessage = styled.article`
  background-color: ${props => props.theme.messageBackground};
  width: 40%;
  margin-top: 10px;
  padding: 10px;
  min-height: 10%;
  max-height: 20%;
  color: white;
  border-radius: ${props =>
    props.type === "send" ? `30px 0px 30px 30px` : `0px 30px 30px 30px`};
  margin-right: ${props => (props.type === "send" ? `20px` : `0px`)};
  margin-left: ${props => (props.type === "receive" ? `20px` : `0px`)};
`;
export default ChatMessage;
