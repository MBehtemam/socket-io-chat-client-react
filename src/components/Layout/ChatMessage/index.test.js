import React from "react";
import styled from "styled-components";
import renderer from "react-test-renderer";
import "jest-styled-components";
import ChatMessage from "./";

describe("Test suits for ChatMessage", () => {
  test("that match to snapshot", () => {
    const tree = renderer.create(<ChatMessage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("that has correct margin-right rule", () => {
    const tree = renderer.create(<ChatMessage type="send" />).toJSON();
    expect(tree).toHaveStyleRule("margin-right", "20px");
  });
  test("that has correct margin-left rule", () => {
    const tree = renderer.create(<ChatMessage type="receive" />).toJSON();
    expect(tree).toHaveStyleRule("margin-left", "20px");
  });
});
