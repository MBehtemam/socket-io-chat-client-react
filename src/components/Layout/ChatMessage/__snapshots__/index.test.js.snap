// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suits for ChatMessage that match to snapshot 1`] = `
.c0 {
  width: 40%;
  margin-top: 10px;
  padding: 10px;
  min-height: 10%;
  max-height: 20%;
  color: white;
  border-radius: 0px 30px 30px 30px;
  margin-right: 0px;
  margin-left: 0px;
}

<article
  className="c0"
/>
`;
